File: java-server/protocol/src/main/java/com/mir2/protocol/MessageCodec.java
package com.mir2.protocol;

/** Client-facing message codec: packed record then legacy 6-bit transport encoding. */
public final class MessageCodec { private MessageCodec(){}
 public static String encode(DefaultMessage m){return SixBitCodec.encodeString(m.toBytes());}
 public static DefaultMessage decode(String wire){return DefaultMessage.fromBytes(SixBitCodec.decodeString(wire));}
}

File: mir2-front/bridge/src/protocol/MessageCodec.ts
import { DefaultMessage } from './DefaultMessage.js';
import { SixBitCodec } from './SixBitCodec.js';

/**
 * Client-facing message codec: packed record then legacy 6-bit transport encoding (16 bytes).
 */
export class MessageCodec {
  private constructor() {}

  public static encode(m: DefaultMessage): string {
    return SixBitCodec.encodeString(m.toBytes());
  }

  public static decode(wire: string): DefaultMessage {
    return DefaultMessage.fromBytes(SixBitCodec.decodeString(wire));
  }
}
